server/balancer.go

```diff
@@ -14,241 +14,208 @@
 package server
 
 import (
-	"github.com/juju/errors"
 	"github.com/ngaut/log"
 	"github.com/pingcap/kvproto/pkg/metapb"
 )
 
-var (
-	_ Balancer = &capacityBalancer{}
-	_ Balancer = &replicaBalancer{}
-	_ Balancer = &leaderBalancer{}
-)
-
-// Balancer is an interface to select store regions for auto-balance.
-type Balancer interface {
-	// Balance selects one store to do balance.
-	Balance(cluster *clusterInfo) (float64, *balanceOperator, error)
-	// GetResourceKind returns resource kind.
-	GetResourceKind() ResourceKind
-}
-
-type capacityBalancer struct {
-	cfg      *BalanceConfig
+type leaderBalancer struct {
+	opt      *scheduleOption
 	kind     ResourceKind
 	selector Selector
 }
 
-func newCapacityBalancer(cfg *BalanceConfig) *capacityBalancer {
+func newLeaderBalancer(opt *scheduleOption) *leaderBalancer {
 	var filters []Filter
-	filters = append(filters, newStateFilter(cfg))
-	filters = append(filters, newCapacityFilter(cfg))
-	filters = append(filters, newSnapCountFilter(cfg))
+	filters = append(filters, newStateFilter(opt))
+	filters = append(filters, newLeaderCountFilter(opt))
 
-	cb := &capacityBalancer{cfg: cfg, kind: storageKind}
-	cb.selector = newBalanceSelector(cb.kind, filters)
-	return cb
+	return &leaderBalancer{
+		opt:      opt,
+		kind:     leaderKind,
+		selector: newBalanceSelector(leaderKind, filters),
+	}
 }
 
-func (cb *capacityBalancer) GetResourceKind() ResourceKind {
-	return cb.kind
+func (l *leaderBalancer) GetName() string {
+	return "leader_balancer"
 }
 
-// Balance tries to select a store region to do balance.
-// The balance type is follower balance.
-func (cb *capacityBalancer) Balance(cluster *clusterInfo) (float64, *balanceOperator, error) {
-	region, source, target := scheduleStorage(cluster, cb.selector)
-	if region == nil {
-		return 0, nil, nil
-	}
+func (l *leaderBalancer) GetResourceKind() ResourceKind {
+	return l.kind
```

Comment:
seem that the kind is fixed for leader and storage balancer, why not returning the const var directly?